Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,46 @@
-Add your answers to the Algorithms exercises here.
\ No newline at end of file
+Add your answers to the Algorithms exercises here.
+
+## Exercise I
+
+Give an analysis of the running time of each snippet of
+pseudocode with respect to the input size n of each of the following:
+
+```
+a)  a = 0 
+    while (a < n * n * n):
+      a = a + n * n
+```
+Exercise I(a) is O(N). It combines a O(0) or O(C) with a while iterator N times that is O(N). The O(C) can be dropped as a worst case leaving it as O(N) time complexity.
+
+```
+b)  sum = 0
+    for i in range(n):
+      i += 1
+      for j in range(i + 1, n):
+        j += 1
+        for k in range(j + 1, n):
+          k += 1
+          for l in range(k + 1, 10 + k):
+            l += 1
+            sum += 1
+```
+Exercise I(b) is O(N^3). It has nested loops which combines 3 for loops based on N inputs and 1 for loop based on a k constant range of 10. O(N) * O(N) * O(N) + O(C) => O(N^3) with the O(C) dropped.
+
```

Comment:
Nice work with these big-O problems.  They are written in a bit of a tricky way and some of these things can be easy to skip over and miss.  Good job!